internal/server/metrics/metrics.go

```diff
@@ -148,6 +139,72 @@ func (m *Metrics) SetMetricByDto(ctx context.Context, d *dto.MetricDTO) error {
 	}
 }
 
+func (m *Metrics) validateMetricDto(d *dto.MetricDTO) error {
+	switch d.MType {
+	case common.CounterMetricType:
+		if d.Delta == nil {
+			return ErrValueParse
```

Comment:
К типизированной ошибке можно добавить сообщение о том, что именно пошло не так. Например:

```suggestion
			return fmt.Errorf("%w: field Delta cannot be nil", ErrValueParse)
```